docs/DATA.md

```diff
@@ -0,0 +1,94 @@
+
+# The Data Layer
+
+Users controlling their own data is a core feature of Habitat. To achieve this, each user's personal node must be capable of serving data in a variety of different formats. Its conceivable that user's will be running applications that simultaneously require relational databases, key-value stores, distributed filesystem access, and more. Habitat refers to these different types of data systems as `datastores`. While these different datastores typically have very different access patterns, Habitat attempts to present a unified data layer that will allow applications to access the datastores through a single API (we need a nice name for this).
+
+```
+Note: This system is probably important enough that we give it its own name, that is not just purely descriptive. 
+```
+
+Why have a single interface for so many different types of datastores? After all, expected data both in and out varies greatly between the different types of datastores. However, there are several major advantages of a single data access layer across all of Habitat:
+1. Interoperable applications are way easier to build if they all use the same API for accessing data.
+2. Consolidating the access point for these datastores reduces the number of ports that have to be opened on each node, simplifying the system.
+3. Common proxying logic can be implemented to forward requests for data not immediately available on the queried node's machine. 
+4. A common authorization and permissions system can be implemented as middleware for this data access API. This also increases the interoperability of applications.
+
+## The API
+
+The data layer is served over `http`. Using the Habitat proxy, HTTP requests can be made over a `LibP2P` transport layer to bypass NATs. All data in the system can be accessed by a unique identifier following this pattern:
+
+unique_identifier: `/<community_id>/<datastore>/<namespace>/<content_identifier>`
```

Comment:
what is the `content_identifier` for relational dbs? it would be `id` right? can we do other types of queries? 

basically is the `content_identifier` supposed to be a way to uniquely identify that piece of data, or a way to specify the query? sometimes these are the same but not necessarily